sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/permissions/PermissionRequestActivity.java

```diff
@@ -0,0 +1,134 @@
+            this.isPermissionGranted = isPermissionGranted;
+            this.exception = exception;
+        }
+    }
+
+    @VisibleForTesting
+    static final String EXTRA_PERMISSIONS = "intent.extra.PERMISSIONS";
+
+    @VisibleForTesting
+    static final int REQUEST_CODE = PermissionRequestActivity.class.getName().hashCode();
+
+    /**
+     * Tracking last request to send result to the listener.
+     */
+    @VisibleForTesting
+    static DefaultAppCenterFuture<Result> sResultFuture;
+
+    /**
+     * Start activity for requesting permissions.
+     *
+     * @param context The context from which the activity will be started.
+     * @param permissions List of requested permissions.
+     * @return Future with the result of a permissions request.
+     */
+    public static AppCenterFuture<Result> requestPermissions(Context context, String... permissions) {
+        if (sResultFuture != null) {
+            AppCenterLog.error(LOG_TAG, "Result future flag is null.");
+            return null;
+        }
+        sResultFuture = new DefaultAppCenterFuture<>();
+        Intent intent = new Intent(context, PermissionRequestActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
+        intent.putExtra(EXTRA_PERMISSIONS, permissions);
+        context.startActivity(intent);
+        return sResultFuture;
+    }
+
+    @VisibleForTesting
+    static void complete(Result result) {
+        if (sResultFuture != null) {
+            sResultFuture.complete(result);
+            sResultFuture = null;
+            return;
+        }
+        AppCenterLog.debug(LOG_TAG, "The start of the activity was not called using the requestPermissions function or the future has already been completed");
+    }
+
+    @Nullable
+    private String[] getPermissionsList() {
+        Intent intent = getIntent();
+        if (intent == null) {
+            return null;
+        }
+        Bundle extras = intent.getExtras();
+        if (extras == null) {
+            return null;
+        }
+        return extras.getStringArray(EXTRA_PERMISSIONS);
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            Exception exception = new Exception("There is no need to request permissions in runtime on Android earlier than 6.0.");
+            AppCenterLog.error(LOG_TAG, "Android version incompatible.", exception);
+            complete(new Result(false, exception));
+            finish();
+            return;
+        }
+        String[] permissions = getPermissionsList();
+        if (permissions == null) {
+            Exception exception = new Exception("Error while getting permissions list from intents extras.");
```

Comment:
Probably better "Failed to get permissions...", as there is no evidence it was an error.